pkg/scheduler/cache/cache.go

```diff
@@ -295,6 +295,13 @@ func (su *defaultStatusUpdater) UpdatePodGroup(pg *schedulingapi.PodGroup) (*sch
 		return nil, err
 	}
 
+	older, err := su.vcclient.SchedulingV1beta1().PodGroups(podgroup.Namespace).Get(context.TODO(), podgroup.Name, metav1.GetOptions{})
```

Comment:
Why do we get the version number before updating? Shouldn't we get the new version number and update again after updating the error?

Reply:
@calvin0327 
I read the description of your answer, shoud we use this
```go 
// IsResourceExpired is true if the error indicates the resource has expired and the current action is
// no longer possible.
// It supports wrapped errors and returns false when the error is nil.
func IsResourceExpired(err error) bool {
	return ReasonForError(err) == metav1.StatusReasonExpired
}
```

Reply:
But I understand that the probability of version conflict is very small. I can't tell whether this error is accidental or will inevitably occur in certain scenarios.

Reply:
I think this a accidental error, if resource version is not change it's work normally

Reply:
> @calvin0327 I read the description of your answer, shoud we use this
> 
> ```go
> // IsResourceExpired is true if the error indicates the resource has expired and the current action is
> // no longer possible.
> // It supports wrapped errors and returns false when the error is nil.
> func IsResourceExpired(err error) bool {
> 	return ReasonForError(err) == metav1.StatusReasonExpired
> }
> ```

the err should be a type `metav1.StatusReasonConflict`.